winforms/tests_backend/widgets/textinput.py

```diff
@@ -1,7 +1,66 @@
-import System.Windows.Forms
+import ctypes
+from ctypes import c_uint
+from ctypes.wintypes import HWND, LPARAM
+
+from System.Windows.Forms import SendKeys, TextBox
 
 from .base import SimpleProbe
+from .properties import toga_xalignment
 
 
 class TextInputProbe(SimpleProbe):
-    native_class = System.Windows.Forms.TextBox
+    native_class = TextBox
+    background_supports_alpha = False
+
+    @property
+    def value(self):
+        return self._placeholder if self.placeholder_visible else self.native.Text
+
+    @property
+    def _placeholder(self):
+        buffer = ctypes.create_unicode_buffer(1024)
+        result = ctypes.windll.user32.SendMessageW(
+            HWND(self.native.Handle.ToInt32()),
+            c_uint(0x1502),  # EM_GETCUEBANNER
+            buffer,
+            LPARAM(ctypes.sizeof(buffer)),
+        )
+        if not result:
+            raise RuntimeError("EM_GETCUEBANNER failed")
+        return buffer.value
+
+    @property
+    def placeholder_visible(self):
+        return not self.native.Text
+
+    @property
+    def placeholder_hides_on_focus(self):
+        False
+
+    @property
+    def readonly(self):
+        return self.native.ReadOnly
+
+    async def type_character(self, char):
+        try:
+            key_code = {
+                "<esc>": "{ESC}",
+                "\n": "{ENTER}",
+            }[char]
+        except KeyError:
+            assert len(char) == 1, char
+            key_code = char
+
+        SendKeys.SendWait(key_code)
+
+    @property
+    def alignment(self):
+        return toga_xalignment(self.native.TextAlign)
+
+    def assert_height(self, min_height, max_height):
+        # Height isn't configurable in this native widget.
+        assert 12 <= self.height <= 22
```

Comment:
The native TextBox seems to ignore its height allocation and always stay the same height. Does this happen with any other widgets? I think I remember us talking about it, but I can't find any similar implementations of this method.

Reply:
There's some similar behavior with Button. Buttons don't expand to fill the available vertical space unless they're *explicitly* set to a given height. 

There's also some similar (but inconsistent) behavior with Switch. On most platforms, the switch "graphic" is a fixed height, regardless of the containing box; however, GTK and iOS both violate this, and render the switch with a weird aspect ratio.

In this case, I'm happy to call this a platform eccentricity and document it.